pandas/tests/frame/methods/test_append.py

```diff
@@ -183,6 +180,9 @@ def test_append_dtypes(self):
         expected = DataFrame(
             {"bar": Series([np.nan, Timestamp("20130101")], dtype="M8[ns]")}
         )
+        if using_array_manager:
+            # With ArrayManager, all-NaN float is not ignored
```

Comment:
can the new JoinUnit.is_valid_na_for be used for these?

Reply:
i think weve landed on the idea being to keep the behaviors matching wherever feasible

Reply:
> I think weve landed on the idea being to keep the behaviors matching wherever feasible

In general where it is easy to do so, yes. But for now I kept the new implementation strictly dtype-dependent without any value-dependent behaviour. I don't think we need to mimic every special case or inconsistency of the BM in the new code, but of course in this case it depends on what behaviour we want long term. I opened https://github.com/pandas-dev/pandas/issues/40893 for this.